### PR TITLE
AutoScaling: adding status type

### PR DESCRIPTION
**Pull request by qiffang**
### What problem does this PR solve? 
Adding autoscaling status type, it includes these item:

- Name // name is used to calculate autoscaling. For example: cpu.

- CurrentValue // the current value of this metric.

- TargetValue // the expect value of this metric.

- CurrentReplicas // the current replica of the component.

- DesiredReplicas // the expect replica of the component.

Code changes

 - Has Go code change

### Does this PR introduce a user-facing change?:
```release-note
NONE
```

### CODE REVIEW

File: pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go [hunk 1]
```diff
@@ -149,6 +149,43 @@ type TidbMonitorRef struct {
 	Name string `json:"name"`
 }
 
-// TODO: sync status
+// BasicAutoScalerSpec describes the basic status for auto-scaling
+type BaseAutoScalerStatus struct {
+	// name is the name of the given metric
+	Name string `json:"name,omitempty"`
```
Comment (Yisaer) [inline pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go]: The metrics property in AutoScaler is a list. If we need to show the metrics value, we should make the metrics status as list.

File: pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go [hunk 2]
```diff
@@ -149,6 +149,43 @@ type TidbMonitorRef struct {
...
+	// +optional
+	LastScaleTime string `json:"lastScaleTime,omitempty" protobuf:"bytes,2,opt,name=lastScaleTime"`
+
+	// currentReplicas is current number of replicas of pods managed by this autoscaler,
+	// as last seen by the autoscaler.
+	CurrentReplicas int32 `json:"currentReplicas" protobuf:"varint,3,opt,name=currentReplicas"`
+
+	// desiredReplicas is the desired number of replicas of pods managed by this autoscaler,
+	// as last calculated by the autoscaler.
+	DesiredReplicas int32 `json:"desiredReplicas" protobuf:"varint,4,opt,name=desiredReplicas"`
```
Comment (Yisaer) [inline pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go]: I recommended name this property as recommendedReplicas.

File: pkg/autoscaler/autoscaler/tidb_autoscaler.go [hunk 3]
```diff
@@ -71,6 +71,7 @@ func syncTiDBAfterCalculated(tc *v1alpha1.TidbCluster, tac *v1alpha1.TidbCluster
 func updateTcTiDBIfScale(tc *v1alpha1.TidbCluster, tac *v1alpha1.TidbClusterAutoScaler, recommendedReplicas int32) error {
 	tac.Annotations[label.AnnTiDBLastAutoScalingTimestamp] = fmt.Sprintf("%d", time.Now().Unix())
 	tc.Spec.TiDB.Replicas = recommendedReplicas
+	tac.Status.TikvStatus.LastScaleTime = time.Now().Format(time.RFC3339)
```
Comment (Yisaer) [inline pkg/autoscaler/autoscaler/tidb_autoscaler.go]: We should record tidb status here.

File: pkg/autoscaler/autoscaler/calculate/cpu.go [hunk 4]
```diff
@@ -15,6 +15,7 @@ package calculate
 
 import (
 	"fmt"
+	"k8s.io/klog"
```
Comment (Yisaer) [inline pkg/autoscaler/autoscaler/calculate/cpu.go]: format goimports

File: pkg/autoscaler/autoscaler/calculate/cpu.go [hunk 5]
```diff
@@ -65,6 +66,24 @@ func CalculateRecomendedReplicasByCpuCosts(tac *v1alpha1.TidbClusterAutoScaler,
 	if err != nil {
 		return -1, err
 	}
+
+	switch memberType {
+	case v1alpha1.TiDBMemberType:
+		tac.Status.TiDBStatus.Name = string(MetricTypeCPU)
+		tac.Status.TiDBStatus.CurrentValue = cpuSecsTotal
+		tac.Status.TiDBStatus.TargetValue = expectedCpuSecsTotal
```
Comment (Yisaer) [inline pkg/autoscaler/autoscaler/calculate/cpu.go]: I think the target Value here should be the average Utilization set by user in autoscaler spec, and the currentValue should be the relevant cpu usage ratio.

File: pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go (hunk 2 above)
Comment (qiffang) [reply pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go]: Accept

File: pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go (hunk 1 above)
Comment (qiffang) [reply pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go]: Currently, We use one metric to calculate the autoscaling. So i think it is just show the metric in effect.

File: pkg/autoscaler/autoscaler/tidb_autoscaler.go (hunk 3 above)
Comment (qiffang) [reply pkg/autoscaler/autoscaler/tidb_autoscaler.go]: Accept

File: pkg/autoscaler/autoscaler/calculate/cpu.go (hunk 5 above)
Comment (qiffang) [reply pkg/autoscaler/autoscaler/calculate/cpu.go]: Accept

**Comment by Yisaer** [comment]
supported in https://github.com/pingcap/tidb-operator/pull/2182

**Pull request closed without merging**